### PR TITLE
Add cognito integration

### PR DESCRIPTION
**Pull request by eddumelendez**
See gh-123

### CODE REVIEW

**Comment by eddumelendez** [comment]
Doc related https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html#amazon-cognito-user-pools-using-the-id-token

File: spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/security/CognitoAuthenticationAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		return jwtDecoder;
+	}
+
+	static class OnUserPoolIdAndRegionPropertiesCondition extends AllNestedConditions {
+
+		OnUserPoolIdAndRegionPropertiesCondition() {
+			super(ConfigurationPhase.PARSE_CONFIGURATION);
+		}
+
+		@ConditionalOnProperty("spring.cloud.aws.security.cognito.userPoolId")
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/security/CognitoAuthenticationAutoConfiguration.java:91]: Properties should not be referenced with camel case. Here: `spring.cloud.aws.security.cognito.user-pool-id`

**Review by maciejwalkowiak — COMMENTED** [review]
If I got it right, with this integration we are saving user from setting Spring Security OAuth config like explained here: https://www.baeldung.com/spring-security-oauth-cognito - is this correct?

I am wondering if we could do more - for example map Cognito groups to GrantedAuthorities (I am not even sure if this is a good idea)

**Comment by eddumelendez** [comment]
> If I got it right, with this integration we are saving user from setting Spring Security OAuth config like explained here: https://www.baeldung.com/spring-security-oauth-cognito - is this correct?

Yes, that's right

> I am wondering if we could do more - for example map Cognito groups to GrantedAuthorities (I am not even sure if this is a good idea)

Not sure either. We can receive feedback once integrated

**Comment by eddumelendez** [comment]
I have tested it and it is working after some polish

**Review by maciejwalkowiak — APPROVED** [review]

**Pull request merged**